runtime/src/accounts_db.rs

```diff
@@ -702,9 +736,13 @@ impl AccountsDB {
         accounts_scan.stop();
 
         let mut clean_old_rooted = Measure::start("clean_old_roots");
-        if !purges_in_root.is_empty() {
-            self.clean_old_rooted_accounts(purges_in_root, max_clean_root);
-        }
+        let (purged_account_slots, removed_accounts) = {
+            if !purges_in_root.is_empty() {
+                self.clean_old_rooted_accounts(purges_in_root, max_clean_root)
+            } else {
+                (HashMap::new(), HashMap::new())
+            }
```

Comment:
nits: how about moving this `is_empty` check inside `clean_old_rooted_accounts` like https://github.com/solana-labs/solana/pull/12606/files#diff-2099c5256db4eb5975c8834af38f6456R888?